osu.Game.Rulesets.Osu/Mods/OsuModBarrelRoll.cs

```diff
@@ -2,52 +2,18 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System.Collections.Generic;
-using osu.Framework.Bindables;
-using osu.Framework.Extensions;
-using osu.Framework.Graphics;
-using osu.Game.Configuration;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Osu.Objects;
 using osu.Game.Rulesets.Osu.Objects.Drawables;
 using osu.Game.Rulesets.Osu.UI;
-using osu.Game.Rulesets.UI;
 using osuTK;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModBarrelRoll : Mod, IUpdatableByPlayfield, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToDrawableHitObjects
+    public class OsuModBarrelRoll : ModBarrelRoll<OsuHitObject>, IApplicableToDrawableHitObjects
     {
-        private float currentRotation;
-
-        [SettingSource("Roll speed", "Rotations per minute")]
-        public BindableNumber<double> SpinSpeed { get; } = new BindableDouble(0.5)
-        {
-            MinValue = 0.02,
-            MaxValue = 12,
-            Precision = 0.01,
-        };
-
-        [SettingSource("Direction", "The direction of rotation")]
-        public Bindable<RotationDirection> Direction { get; } = new Bindable<RotationDirection>(RotationDirection.Clockwise);
-
-        public override string Name => "Barrel Roll";
-        public override string Acronym => "BR";
-        public override string Description => "The whole playfield is on a wheel!";
-        public override double ScoreMultiplier => 1;
-
-        public override string SettingDescription => $"{SpinSpeed.Value} rpm {Direction.Value.GetDescription().ToLowerInvariant()}";
-
-        public void Update(Playfield playfield)
-        {
-            playfield.Rotation = currentRotation = (Direction.Value == RotationDirection.Counterclockwise ? -1 : 1) * 360 * (float)(playfield.Time.Current / 60000 * SpinSpeed.Value);
-        }
-
-        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
-        {
-            // scale the playfield to allow all hitobjects to stay within the visible region.
-            drawableRuleset.Playfield.Scale = new Vector2(OsuPlayfield.BASE_SIZE.Y / OsuPlayfield.BASE_SIZE.X);
-        }
+        protected override Vector2 PlayfieldScale => new Vector2(OsuPlayfield.BASE_SIZE.Y / OsuPlayfield.BASE_SIZE.X);
```

Comment:
This can probably be automatically determined by checking the `Playfield`'s `DrawSize`.

Reply:
Yep, seems to be working